packages/vite/src/node/plugins/resolve.ts

```diff
@@ -172,8 +173,7 @@ export function resolvePlugin(resolveOptions: InternalResolveOptions): Plugin {
       // relative
       if (
         id.startsWith('.') ||
-        (preferRelative && /^\w/.test(id)) ||
-        importer?.endsWith('.html')
+        ((preferRelative || importer?.endsWith('.html')) && /^\w/.test(id))
```

Comment:
I think this was incorrect. If importer is html it was always handled as relative path.

So `/foo.svelte` will be resolved to
```js
const basedir = importer ? path.dirname(importer) : process.cwd() // C:\Users\green\workspace
const fsPath = path.resolve(basedir, id) // C:\foo.svelte
const normalizedFsPath = normalizePath(fsPath) // C:/foo.svelte
```